Detectors/DCS/include/DetectorsDCS/DCSProcessor.h

```diff
@@ -0,0 +1,143 @@
+
+  template<typename T>
+  int processArrayType(const std::vector<DPID>& array, DeliveryType type, const std::unordered_map<DPID, DPVAL>& map, std::vector<uint64_t>& latestTimeStamp, std::unordered_map<DPID, T>& destmap);
+
+  std::unordered_map<DPID, DPVAL>::const_iterator processAlias(const DPID& alias, DeliveryType type, const std::unordered_map<DPID, DPVAL>& map);
+
+  void processChars();
+  void processInts();
+  void processDoubles();
+  void processUInts();
+  void processBools();
+  void processStrings();
+  void processTimes();
+  void processBinaries();
+  uint64_t processFlag(uint64_t flag, const char* alias);
+
+  void doSimpleMovingAverage(int nelements, std::deque<int>& vect, float& avg, bool& isSMA);
+
+  DQChars& getVectorForAliasChar(const DPID& id) { return mDpscharsmap[id]; }
+  DQInts& getVectorForAliasInt(const DPID& id) { return mDpsintsmap[id]; }
+  DQDoubles& getVectorForAliasDouble(const DPID& id) { return mDpsdoublesmap[id]; }
+  DQUInts& getVectorForAliasUInt(const DPID& id) { return mDpsUintsmap[id]; }
+  DQBools& getVectorForAliasBool(const DPID& id) { return mDpsboolsmap[id]; }
+  DQStrings& getVectorForAliasString(const DPID& id) { return mDpsstringsmap[id]; }
+  DQTimes& getVectorForAliasTime(const DPID& id) { return mDpstimesmap[id]; }
+  DQBinaries& getVectorForAliasBinary(const DPID& id) { return mDpsbinariesmap[id]; }
+
+ private:
+  float mAvgTestInt0 = 0; // moving average for DP named TestInt0
+  std::unordered_map<DPID, DQChars> mDpscharsmap;
+  std::unordered_map<DPID, DQInts> mDpsintsmap;
+  std::unordered_map<DPID, DQDoubles> mDpsdoublesmap;
+  std::unordered_map<DPID, DQUInts> mDpsUintsmap;
+  std::unordered_map<DPID, DQBools> mDpsboolsmap;
+  std::unordered_map<DPID, DQStrings> mDpsstringsmap;
+  std::unordered_map<DPID, DQTimes> mDpstimesmap;
+  std::unordered_map<DPID, DQBinaries> mDpsbinariesmap;
+  std::vector<DPID> mAliaseschars;
+  std::vector<DPID> mAliasesints;
+  std::vector<DPID> mAliasesdoubles;
+  std::vector<DPID> mAliasesUints;
+  std::vector<DPID> mAliasesbools;
+  std::vector<DPID> mAliasesstrings;
+  std::vector<DPID> mAliasestimes;
+  std::vector<DPID> mAliasesbinaries;
+  std::vector<uint64_t> mLatestTimestampchars;
+  std::vector<uint64_t> mLatestTimestampints;
+  std::vector<uint64_t> mLatestTimestampdoubles;
+  std::vector<uint64_t> mLatestTimestampUints;
+  std::vector<uint64_t> mLatestTimestampbools;
+  std::vector<uint64_t> mLatestTimestampstrings;
+  std::vector<uint64_t> mLatestTimestamptimes;
+  std::vector<uint64_t> mLatestTimestampbinaries;
+
+  ClassDefNV(DCSProcessor, 0);
+};
+
+  using Ints = std::vector<int>;
+  using Chars = std::vector<char>;
+  using Doubles = std::vector<double>;
+  using Binaries = std::array<uint64_t, 7>;
+  using Strings = std::array<char, 56>;
+
+  using DQStrings = std::deque<Strings>;
+  using DQBinaries = std::deque<Binaries>;
+
+  using DPID = o2::dcs::DataPointIdentifier;
+  using DPVAL = o2::dcs::DataPointValue;
```

Comment:
you don't need to redefine these, just use below e.g. ``DCSProcessor::DQStrings`` instead of ``DQStrings``

Reply:
Yes, after making the public, it works of course.